src/BotService/Infrastructure/Core/CallHandler.cs

```diff
@@ -275,6 +288,85 @@ private static StartStreamExtractionResponse GetStreamExtractionResponse(Protoco
             }
         }
 
+        private int AssignExtractionPort(StartStreamExtractionBody streamBody)
+        {
+            int port = 0;
+
+            switch (streamBody.Protocol)
+            {
+                case Protocol.SRT:
+                    if (!_availablePorts.TryTake(out int srtPort))
+                    {
+                        _logger.LogWarning("[Call Handler] There are no ports available");
+
+                        throw new StartStreamExtractionException("There are no ports available");
+                    }
+
+                    port = srtPort;
+                    break;
+                case Protocol.RTMP:
+                    var rtmpStreamBody = (RtmpStreamExtractionBody)streamBody;
+                    int rtmpPort = 0;
+
+                    if (rtmpStreamBody.EnableSsl)
+                    {
+                        if (!_availableRtmpsPorts.TryTake(out int rtmps))
+                        {
+                            _logger.LogWarning("[Call Handler] There are no ports available");
+
+                            throw new StartStreamExtractionException("There are no ports available");
+                        }
+
+                        rtmpPort = rtmps;
+                    }
+                    else
+                    {
+                        if (!_availableRtmpPorts.TryTake(out int rtmp))
+                        {
+                            _logger.LogWarning("[Call Handler] There are no ports available");
+
+                            throw new StartStreamExtractionException("There are no ports available");
+                        }
+
+                        rtmpPort = rtmp;
+                    }
+
+                    port = rtmpPort;
+                    break;
+                default:
+                    throw new ArgumentException("Protocol not supported", nameof(streamBody));
+            }
+
+            _currentAssignedPorts.AddOrUpdate(streamBody.ParticipantGraphId, port, (k, v) => port);
+
+            return port;
+        }
+
+        private void RemoveAssignedExtractionPort(string streamId, Protocol protocol)
+        {
+            _currentAssignedPorts.TryRemove(streamId, out int port);
+
+            switch (protocol)
+            {
+                case Protocol.SRT:
+                    _availablePorts.Add(port);
+                    break;
+                case Protocol.RTMP:
+                    if (port >= DefaultRtmpPort && port <= (DefaultRtmpPort + _numberOfMultiviewSockets))
```

Comment:
**nitpick(blocking)**: we must modify this logic based on previous comments. RTMP --> 19xx, RTMPS --> 29xx